core/src/main/java/com/linecorp/armeria/common/FlagsProvider.java

```diff
@@ -581,6 +581,22 @@ default Integer defaultMaxClientNumRequestsPerConnection() {
         return null;
     }
 
+    /**
+     * Returns the default client-side graceful connection shutdown timeout in microseconds.
+     *
+     * <p>Note that this flag has no effect if a user specified the value explicitly via
+     * {@link ClientFactoryBuilder#http2GracefulShutdownTimeoutMillis(long)}.
+     *
+     * <p>The default value of this flag is
+     * {@value DefaultFlagsProvider#DEFAULT_CLIENT_HTTP2_GRACEFUL_SHUTDOWN_TIMEOUT_MILLIS}.
+     * Specify the {@code -Dcom.linecorp.armeria.defaultClientHttp2GracefulShutdownTimeoutMillis=<long>}
+     * JVM option to override the default value. {@code 0} disables the graceful shutdown.
```

Comment:
Value of 0 will not disable the graceful shutdown because below logic in `Http2ClientConnectionHandlerBuilder` 
🤔 
```java 
        final long gracefulShutdownTimeoutMillis = clientFactory.http2GracefulShutdownTimeoutMillis();
        if (gracefulShutdownTimeoutMillis == Long.MAX_VALUE) {
            gracefulShutdownTimeoutMillis(-1);
        } else {
            gracefulShutdownTimeoutMillis(gracefulShutdownTimeoutMillis);
        }
```

Reply:
Should we change the docs to `-1` ?

Reply:
We don't accept the value `-1` conventionally. 😉  
If the value is `0`, `doClose()` is called right away so there's no graceful shutdown:
https://github.com/netty/netty/blob/4.1/codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java#L982

Reply:
Ah ~ thank you for the clarification 👍